llvm/lib/Target/AArch64/MachineSMEABIPass.cpp

```diff
@@ -200,7 +200,7 @@ struct MachineSMEABI : public MachineFunctionPass {
 
   /// Inserts code to handle changes between ZA states within the function.
   /// E.g., ACTIVE -> LOCAL_SAVED will insert code required to save ZA.
-  void insertStateChanges();
+  void insertStateChanges(bool IsAgnosticZA);
```

Comment:
nit: Rather than passing this everywhere, maybe just cache this locally in the `MachineSMEABI` struct?

Reply:
I'm just replaced this with `AFI->getSMEFnAttrs().hasAgnosticZAInterface()` rather than having the separate `IsAgnosticZA` parameter. I don't think there will be any other cases that need handling in the near future.